iothub_client/src/iothub_client_core.c

```diff
@@ -860,9 +854,17 @@ static IOTHUB_CLIENT_RESULT StartWorkerThreadIfNeeded(IOTHUB_CLIENT_CORE_INSTANC
     }
     else
```

Comment:
So the change here is to start both threads? for the regular client and for the transport?
I am not sure this correct. We should discuss this solution.